### PR TITLE
Refactoring to avoid storing tokens and load groups correctly

### PR DESCRIPTION
**Pull request by Erwyn**
This is a part of [story #14018](https://tuleap.net/plugins/tracker/?aid=14018) have a central management of users and groups using oauth

* Update of tuleap-api dependency to v1.2.0

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java [hunk 1]
```diff
@@ -176,27 +184,56 @@ private void setClientSecret(String secretString) {
...
+        final TuleapAuthenticationToken tuleapAuthenticationToken = (TuleapAuthenticationToken) authenticatedUserAcegiToken;
+
+        if (this.tokenHasTuleapGroup(tuleapAuthenticationToken, groupName)) {
+            return new TuleapGroupDetails(groupName);
+        }
+
+        throw new UsernameNotFoundException("Group " + groupName + " not found for current Tuleap user");
+    }
+
+    private Boolean tokenHasTuleapGroup(TuleapAuthenticationToken token, String groupName) {
```
Comment (robinsoc) [inline src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java:229]: what do you think about hasTuleapGroupToken instead?

File: src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java [hunk 2]
```diff
@@ -176,27 +184,56 @@ private void setClientSecret(String secretString) {
...
+        }
+
+        final TuleapAuthenticationToken tuleapAuthenticationToken = (TuleapAuthenticationToken) authenticatedUserAcegiToken;
+
+        if (this.tokenHasTuleapGroup(tuleapAuthenticationToken, groupName)) {
+            return new TuleapGroupDetails(groupName);
+        }
+
+        throw new UsernameNotFoundException("Group " + groupName + " not found for current Tuleap user");
+    }
```
Comment (robinsoc) [inline src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java:227]: To be consistent with loadByUsername(), is it possible to have the return at the end of the method ?

File: src/main/java/io/jenkins/plugins/tuleap_oauth/helper/PluginHelper.java [hunk 3]
```diff
@@ -3,10 +3,14 @@
 import com.auth0.jwk.InvalidPublicKeyException;
 import com.auth0.jwk.Jwk;
 import com.auth0.jwt.algorithms.Algorithm;
+import hudson.model.User;
+import io.jenkins.plugins.tuleap_oauth.TuleapAuthenticationToken;
```
Comment (robinsoc) [inline src/main/java/io/jenkins/plugins/tuleap_oauth/helper/PluginHelper.java:7]: Please remove. It seems it's not used

File: src/main/java/io/jenkins/plugins/tuleap_oauth/helper/PluginHelper.java [hunk 4]
```diff
@@ -3,10 +3,14 @@
...
 import io.jenkins.plugins.tuleap_server_configuration.TuleapConfiguration;
 import jenkins.model.Jenkins;
 import okhttp3.Response;
 import okhttp3.ResponseBody;
+import org.acegisecurity.Authentication;
+import org.acegisecurity.context.SecurityContext;
```
Comment (robinsoc) [inline src/main/java/io/jenkins/plugins/tuleap_oauth/helper/PluginHelper.java:13]: Please remove. It seems it's not used

File: src/main/java/io/jenkins/plugins/tuleap_oauth/helper/PluginHelperImpl.java [hunk 5]
```diff
@@ -3,10 +3,14 @@
 import com.auth0.jwk.InvalidPublicKeyException;
 import com.auth0.jwk.Jwk;
 import com.auth0.jwt.algorithms.Algorithm;
+import hudson.model.User;
+import io.jenkins.plugins.tuleap_oauth.TuleapAuthenticationToken;
```
Comment (robinsoc) [inline src/main/java/io/jenkins/plugins/tuleap_oauth/helper/PluginHelperImpl.java:7]: Please remove. It seems it's not used

File: src/test/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealmTest.java [hunk 6]
```diff
@@ -157,4 +166,128 @@ public void testTheValidationIsNotOkWhenTheClientIdIsEmpty() {
...
+        tuleapSecurityRealm.loadUserByUsername(username);
+    }
+
+
+
+
+
+
+
+
```
Comment (robinsoc) [inline src/test/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealmTest.java:235]: :open_mouth:

File: src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java (hunk 1 above)
Comment (Erwyn) [reply src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java:229]: If your not against keeping it, I would rather not to. I find it very much easier to read the sentence "if token has tuleap group" than "if has tuleap group token"

File: src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java (hunk 2 above)
Comment (Erwyn) [reply src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java:227]: sure

File: src/main/java/io/jenkins/plugins/tuleap_oauth/helper/PluginHelper.java (hunk 3 above)
Comment (Erwyn) [reply src/main/java/io/jenkins/plugins/tuleap_oauth/helper/PluginHelper.java:7]: done

File: src/test/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealmTest.java (hunk 6 above)
Comment (Erwyn) [reply src/test/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealmTest.java:235]: ah yes I know where this comes from

**Pull request merged**